source/_docs/deprecated_constructor_notices.md

```diff
@@ -0,0 +1,52 @@
+categories: [developing]
+tags: [code, php]
+keywords: debug, php 7, php error, php errors, php fpm error log, fpm error log, php log, php execution, execute php, caching, cache, php exception, exceptions, errors
+---
+
+PHP notices are usually handled automatically by the Pantheon Platform as described on the page [PHP Errors and Exceptions](/docs/php-errors); however, occasionally a PHP notice might be emitted directly into the web page content.
+
+An example notice might look like this:
+```nohighlight
+Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; views_display has a deprecated constructor in /srv/bindings/46027a30ca4c4980a7188036eb2fcea5/code/sites/all/modules/views/includes/view.inc on line 2553 
+Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; views_many_to_one_helper has a deprecated constructor in /srv/bindings/46027a30ca4c4980a7188036eb2fcea5/code/sites/all/modules/views/includes/handlers.inc on line 753
+```
+
+When an error like this is encountered, it will usually only appear intermitantly; most page loads will come up correctly until, after a period of inactivity on the site, the same message will appear again.
+
+This error is caused by a combination of two factors:
+
+- PHP notices generated during source code parsing
+- PHP opcache is in use
+
+Starting with PHP 7, a [deprecation notice for PHP-4-style constructors](http://php.net/manual/en/migration70.deprecated.php#migration70.deprecated.php4-constructors) is emitted if a method with the same name as its class is used. This was the recommended way to declare class constructors in PHP 4, but it was not until PHP 7 that this form actually started producing a notice. At this time, the PHP-4-style constructor is the only deprecation notice that is reported during source code parsing. In theory, this structure should only be encountered in very old code; in practice, though, some projects may have continued using the deprecated form, so it may be encountered from time to time.
+
+Opcache is always enabled on Pantheon. When opcache is in use, PHP will print any notice emitted during source code processing directly to the standard output, ignoring any error handler that may be set. These notices may still be disabled by setting `error_reporting` to ignore E_DEPRECATED; however, Drupal always enables E_DEPRECATED in `error_reporting` early in its bootstrap process, so these notices will be printed if `display_errors` is set to `on`.
+
+Opcache also influences when this error is displayed. Since opcache caches the compiled form (opcodes) of the PHP that is being executed, the deprecation notices will not be printed when the PHP opcodes are fetched from the opcache. This is what leads to the intermittant nature of this problem. To make it easier to find and debug PHP 4 constructors, you may disable the opcache in your settings.php file:
```

Comment:
and keep/tweak this line: 

To make it easier to find and debug PHP 4 constructors, **temporarily** disable the opcache in your settings.php file: